### PR TITLE
New functionality to rename and delete player

### LINKED ISSUE #12: REQ-011: Player Management
# Player Management System Implementation

## Description
Implement functionality to delete players by ID and modify player nicknames, maintaining list order and data persistence.

### Required Functions
```c
void deletePlayerById(Player** head, int id);
void modifyPlayerNickname(Player* head, int id, char* newNickname);
```

### Interface Format
```
===============================
PLAYER MANAGEMENT
===============================
1. Delete Player
2. Modify Nickname
3. Back to Menu

Option: _

[If Delete Selected]
Enter player ID to delete: _

[If Modify Selected]
Enter player ID to modify: _
Enter new nickname: _
```

### Core Features
* Player Deletion:
  - Delete by ID
  - Update doubly-linked list
  - Memory management
  - File update
* Nickname Modification:
  - Validate new nickname
  - Check for duplicates
  - Maintain score
  - Update file

## Acceptance Criteria
- [ ] Deletion functionality:
  - Removes correct player
  - Maintains list integrity
  - Updates double links
  - Frees memory properly
  - Updates file
- [ ] Modification functionality:
  - Updates nickname
  - Validates new nickname
  - Checks duplicates
  - Maintains score
  - Updates file
- [ ] Input validation:
  - Valid ID check
  - Valid nickname format
  - Duplicate nickname check
- [ ] Error handling:
  - ID not found
  - Invalid input
  - Duplicate nicknames
  - File operation errors
- [ ] List integrity:
  - Double links maintained
  - Order preserved
  - File consistency

## Technical Notes
* Handle list edge cases:
  - Empty list
  - Single node
  - First/last node
  - Middle node
* Implement proper memory management
* Validate all user inputs
* Update persistence after changes
* Maintain sorted order if needed

## Integration Points
* Works with Player structure
* Uses file operations
* Maintains data consistency
* Updates player records

### PR DESCRIPTION
**Pull request by Mr-Mauxerr**
Adds two functions, one of them is called "Delete Player" wich is used to delete a player by its id, the another one is called "Rename player" wich, also, by the id is able to rename a player.

### CODE REVIEW

**Review by gfdfranco — APPROVED** [review]
LGTM

**Review by Nohelany — APPROVED** [review]
LGTM

File: marathon_quiz.cpp [hunk 1]
```diff
@@ -452,4 +453,40 @@ void deletePlayer(Player** head){
...
+    }
+
+    //if the id doesnt exist exits
+    if (current == NULL) {
+        printf("There's no such a player with the ID: %d\n", playerId);
+        return;
+    }
+    //if the id exist now it asks you the name
+    printf("How wants this person to be called?:\n ");
+    scanf(" %[^\n]", nickname);
```
Comment (J-P-T-0) [inline marathon_quiz.cpp]: looks good I just would advise to replace this line with the nicknameCreation function, but it isn't necessary

Comment (J-P-T-0) [inline marathon_quiz.cpp]: looks good I just would advise to replace this line with the nicknameCreation function, but it isn't necessary

**Review by J-P-T-0 — CHANGES_REQUESTED** [review]
LGTM, if changes are not necessary

File: marathon_quiz.cpp (hunk 1 above)
Comment (gfdfranco) [reply marathon_quiz.cpp]: Great recommendation @J-P-T-0 , could u please help @Mr-Mauxerr to implement this change, ty !

File: marathon_quiz.cpp [hunk 2]
```diff
@@ -483,10 +482,9 @@ void changeName(Player* head) {
     }
     //if the id exist now it asks you the name
     printf("How wants this person to be called?:\n ");
-    scanf(" %[^\n]", nickname);
+    fgets(nickname, MAX_STRING_NICKNAME, stdin);
```
Comment (J-P-T-0) [inline marathon_quiz.cpp]: the change must be made in this part XD this is the most recent commit sorry

**Review by J-P-T-0 — APPROVED** [review]
Issue is now solved, LGTM

**Pull request merged**